test_scripts/Capabilities/PersistingHMICapabilities/common.lua

```diff
@@ -0,0 +1,909 @@
+  local mobSession = actions.mobile.getSession()
+  local cid = mobSession:SendRPC("GetSystemCapability", { systemCapabilityType = pSystemCapabilityType })
+  mobSession:ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+  :ValidIf(function(_,data)
+    for param, value in pairs (pResponseCapabilities) do
+      if not data.payload.systemCapability[param] then
+        errorMessages = errorMessages ..
+          errorMessage(param, "does not exist", value)
+      else
+        if not utils.isTableEqual(data.payload.systemCapability[param], value) then
+          errorMessages = errorMessages ..
+            errorMessage(param, data.payload.systemCapability[param], value)
+        end
+      end
+    end
+    if string.len(errorMessages) > 0 then
+      return false, errorMessages
+    else
+      return true
+    end
+  end)
+end
+
+function m.changeLanguage(pLanguage)
+  local hmiConnection = actions.hmi.getConnection()
+  hmiConnection:SendNotification("TTS.OnLanguageChange", { language = pLanguage })
+  hmiConnection:SendNotification("VR.OnLanguageChange", { language = pLanguage })
+  hmiConnection:SendNotification("UI.OnLanguageChange", { language = pLanguage })
+end
+
+function m.checkLanguageCapability(pLanguage)
+  local data = SDL.HMICapCache.get()
+  if data and data.VR and data.VR.language == pLanguage
+      and data.TTS and data.TTS.language == pLanguage
+      and data.UI and data.UI.language == pLanguage then
+    utils.cprint(35, "Languages were changed")
+  else
+    actions.run.fail("SDL doesn't update cache file")
+  end
+end
+
+function m.updateHMILanguageCapability(pLanguage)
+  local hmiValues = m.getDefaultHMITable()
+  hmiValues.UI.GetLanguage.params.language = pLanguage
+  hmiValues.VR.GetLanguage.params.language = pLanguage
+  hmiValues.TTS.GetLanguage.params.language = pLanguage
+  return hmiValues
+end
+
+function m.registerApp(pAppId, pCapResponse, pMobConnId, hasPTU)
+  if not pCapResponse then pCapResponse = {} end
+  if not pAppId then pAppId = 1 end
+  if not pMobConnId then pMobConnId = 1 end
+
+  local policyModes = {
+    P  = "PROPRIETARY",
+    EP = "EXTERNAL_PROPRIETARY",
+    H  = "HTTP"
+  }
+
+  local session = actions.mobile.createSession(pAppId, pMobConnId)
+  session:StartService(7)
+  :Do(function()
+      local corId = session:SendRPC("RegisterAppInterface", actions.app.getParams(pAppId))
+      actions.hmi.getConnection():ExpectNotification("BasicCommunication.OnAppRegistered",
+        { application = { appName = actions.app.getParams(pAppId).appName }})
+      :Do(function(_, d1)
+          actions.app.setHMIId(d1.params.application.appID, pAppId)
+          if hasPTU then
+            m.ptu.expectStart()
+          end
+        end)
+      local errorMessages = ""
+      session:ExpectResponse(corId, { success = true, resultCode = "SUCCESS" })
+      :ValidIf(function(_,data)
+        for param, value in pairs (pCapResponse) do
+          if param == "hmiZoneCapabilities" then
```

Comment:
would it be better to check `type(elem) == "table"` versus this?

Reply:
@iCollin You are right. AS I notice the function `utils.isTableEqual()` can compare both tables and simple types, so all mentioned check are replaced next way:
```
if param == "hmiZoneCapabilities" then
  if not data.payload[param] == value then
    errorMessages = errorMessages ..
      errorMessage(param, data.payload[param], value)
  end
else
  if not utils.isTableEqual(data.payload[param], value) then
    errorMessages = errorMessages ..
      errorMessage(param, data.payload[param], value)
  end
end
```
=>
```
if not utils.isTableEqual(data.payload[param], value) then
  errorMessages = errorMessages ..
      errorMessage(param, data.payload[param], value)
end
```

Also was fixed preparation of data related to `hmiZoneCapabilities` parameter of UI capabilities for validation of RAI response from SDL 
(hmiZoneCapabilities parameter is a string in HMI API and it is an array of strings in Mobile API)  

See 12a07e8